### PR TITLE
Fix error reporting when trace data is not included

### PR DESCRIPTION
**Pull request by trevor-scheer**
This is a 2-part change which requires an update to the gateway runtime as well. Related PR: https://github.com/apollographql/federation/pull/2242

It was brought to our attention that trace information w.r.t. subgraph errors is incorrect in the case that there is no trace in the subgraph response (i.e. no ftv1 support in the subgraph or `fieldLevelInstrumentation` is 0 or returns 0).

With this change:
* Errors reported by subgraphs (with no trace data in the response) are now accurately reflected in the numeric error stats
* Operations that receive errors from subgraphs (with no trace data in the response) are no longer sent as incomplete, error-less traces

The gateway change now collects errors separately when no trace data is found and passes that error information (subgraph and error path) along with the other metrics data for the plugin to consume. The plugin uses the subgraph name and error path in order to add error stat information to the tree.

### CODE REVIEW

**Review by glasser — CHANGES_REQUESTED** [review]
The broad structure here seems good but the tests suggest there's a bug preventing the fix from working. (Maybe the version of gateway loaded is wrong?)

Also I'm sure you know this but a ton more comments would be helpful, both about the new stuff and about the other stuff we've discovered today that wasn't obvious.

File: package.json
Comment (trevor-scheer) [reply package.json:42]: The "correct" way to do this would be publish gateway interface -> gateway -> server, but yes it is just tests.

File: packages/gateway-interface/src/index.ts
Comment (trevor-scheer) [reply packages/gateway-interface/src/index.ts]: [7d4804b](https://github.com/apollographql/apollo-server/pull/7136/commits/7d4804b42d49fc72e3b2f609b0c0fa9b06e6e251)

File: packages/gateway-interface/src/index.ts
Comment (glasser) [inline packages/gateway-interface/src/index.ts:121]: unrelated: the comment at the very top of this file is outdated

File: packages/server/src/plugin/usageReporting/plugin.ts
Comment (glasser) [inline packages/server/src/plugin/usageReporting/plugin.ts:741]: (a) this probably deserves a comment (or an expansion of the larger comment above, which maybe could be a bulleted list by now)
(b) might as well do this boolean check before calling the hook on the previous line instead of after?

File: packages/server/src/plugin/usageReporting/stats.ts
Comment (glasser) [inline packages/server/src/plugin/usageReporting/stats.ts:174]: perhaps `GraphQLRequestMetrics['nonFtv1ErrorPaths']` if you don't want to give it a name (and in a couple other places)

**Review by glasser — COMMENTED** [review]
Other than those minor notes this looks good. The PR title/description could use an update to describe this kinda subtle issue, and also a changeset.

File: packages/gateway-interface/src/index.ts
Comment (trevor-scheer) [reply packages/gateway-interface/src/index.ts]: [12355d9](https://github.com/apollographql/apollo-server/pull/7136/commits/12355d927660a24ad744e784158948e63d9c8931)

File: packages/server/src/plugin/usageReporting/stats.ts
Comment (trevor-scheer) [reply packages/server/src/plugin/usageReporting/stats.ts]: 52f8b6494d7009b1dfadf3b4f830854f44f50585

File: packages/server/src/plugin/usageReporting/plugin.ts
Comment (trevor-scheer) [reply packages/server/src/plugin/usageReporting/plugin.ts]: [5c19b5a](https://github.com/apollographql/apollo-server/pull/7136/commits/5c19b5ad6e4e15e2bedb9d86514bb4fce81bb8ec)

File: .changeset/heavy-poets-wink.md
Comment (glasser) [inline .changeset/heavy-poets-wink.md:5]: Does this changeset (and PR description) need to note that a particular version of Gateway is also needed to have the issue be actually fixed?

File: package.json
Comment (glasser) [inline package.json:41]: Would be nice to avoid this but I guess the gateway release process is slow.

File: package.json
Comment (glasser) [inline package.json:43]: Why is this needed?

File: package.json
Comment (trevor-scheer) [reply package.json:41]: This will become the actual gateway version once the patch is released, won't land this until then

File: package.json
Comment (trevor-scheer) [reply package.json:43]: Probably could've always been just the current version of subgraph but this will be latest once the patch release is published

File: .changeset/heavy-poets-wink.md
Comment (trevor-scheer) [reply .changeset/heavy-poets-wink.md:5]: Added in [e1b8122](https://github.com/apollographql/apollo-server/pull/7136/commits/e1b8122a78b3d400b347f612ef967cd179e05cb4) (with a FIXME to plug in final gateway version)

**Pull request merged**
